sycl/include/sycl/types.hpp

```diff
@@ -1544,8 +1544,14 @@ template <typename VecT, typename OperationLeftT, typename OperationRightT,
           template <typename> class OperationCurrentT, int... Indexes>
 class SwizzleOp {
   using DataT = typename VecT::element_type;
-  using CommonDataT = std::common_type_t<typename OperationLeftT::DataT,
-                                         typename OperationRightT::DataT>;
+  // Certain operators return a vector with a different element type. Also, the
+  // left and right operand types may differ. CommonDataT selects a result type
+  // based on these types to ensure that the result value can be represented.
```

Comment:
Honestly, I wanted something more concrete/specific here, maybe with a few example to represent each scenario. The cases when `std::common_type_t` fails are of particular interest.